uitools/cpp/Esri/ArcGISRuntime/Toolkit/FloorFilterController.cpp

```diff
@@ -549,54 +603,28 @@ namespace Toolkit {
    */
   FloorFilterFacilityItem* FloorFilterController::facility(const QString& facilityId) const
   {
-    auto model = m_facilities;
-    const auto rows = model->rowCount();
-    for (int i = 0; i < rows; ++i)
-    {
-      auto index = model->index(i);
-      auto item = model->element<FloorFilterFacilityItem>(index);
-      if (item->modelId() == facilityId)
-        return item;
-    }
-    return nullptr;
+    return findElement<FloorFilterFacilityItem>(m_facilities, facilityId);
   }
 
   /*!
    \brief Returns the item with \c modelId matching \a siteId by searching \l sites.
    */
   FloorFilterSiteItem* FloorFilterController::site(const QString& siteId) const
   {
-    auto model = m_sites;
-    const auto rows = model->rowCount();
-    for (int i = 0; i < rows; ++i)
-    {
-      auto index = model->index(i);
-      auto item = model->element<FloorFilterSiteItem>(index);
-      if (item->modelId() == siteId)
-        return item;
-    }
-    return nullptr;
+    return findElement<FloorFilterSiteItem>(m_sites, siteId);
   }
 
   /*!
    \brief Returns the item with \c modelId matching \a levelId by searching \l levels.
    */
   FloorFilterLevelItem* FloorFilterController::level(const QString& levelId) const
   {
-    auto model = m_levels;
-    const auto rows = model->rowCount();
-    for (int i = 0; i < rows; ++i)
-    {
-      auto index = model->index(i);
-      auto item = model->element<FloorFilterLevelItem>(index);
-      if (item->modelId() == levelId)
-        return item;
-    }
-    return nullptr;
+    return findElement<FloorFilterLevelItem>(m_levels, levelId);
   }
 
   /*!
    \internal
+   \property Esri::ArcGISRuntime::Toolkit::FloorFilterController::selectedFacility
```

Comment:
I'm not sure how this works - marked as internal but also doc'ing it as a property. I see it in a few places. Can you please confirm in the output?

Reply:
You are right, qdoc ignores anything with the `\internal` flag.

But qdoc does also have a flag:

```sh
qdoc --showinternal ...
```

Which reads as:

> Include content marked internal

So, marking as `\property` and `\internal` is not really helpful in the grand scheme of things, but IMO, still good practice, and I suppose someone running `qdoc --showinternal` could _theoretically_ generate some sort of documentation if they really wanted.

Reply:
That's fair comment! Removed `\internal` and added `\brief` tags.